server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -842,6 +854,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             coordinationMetaData = new CoordinationMetaData(in);
             transientSettings = Settings.readSettingsFromStream(in);
             persistentSettings = Settings.readSettingsFromStream(in);
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
Isn't this feature going into 7.2.0? So all of these conditionals should be using that version constant?

Reply:
sure, but I'll change these after I merge the backport of this. Unless I'm missing smth.

Reply:
I think that is an error-prone approach. Instead, you can make these the final version in this PR, and disable bwc here (see the `bwc_tests_enabled` flag in the root `build.gradle`), switching back on in a followup to master. This way only one change needs to be made, versus remembering every conditional modified here. You can still test the bwc locally using the `tests.bwc.refspec.7.x` system property pointing to your local 7.x based branch with the changes backported.

Reply:
I see, thank you for the details!

But I prefer to disable bwc tests and change all the versions in a follow-up, after this has been merged to master. This way, this PR is true to what master will look like in the end. If this approach creates one extra 7.x backport PR and one extra follow-up PR to disable tests and update versions (with "full" CI), then I am personally okay with it. I don't feel confident enough to cut steps in this.

I will go with the presented approach, if you don't mind.

Reply:
But it's *not* true to what master will be, that is the problem. In the end, the conditionals will use the 7.3.0 version (not the change since 7.2 was branched). The flag I mentioned exists for this explicit purpose: to allow checking into master the exact code that will be backported.

Reply:
Alright, I have acted on the suggestion to set the 7.3 versions and disable bwc.